### PR TITLE
refactor: modularize carousel wiring

### PR DESCRIPTION
**Pull request by CyanAutomation**
### Task Contract
```json
{
  "inputs": [
    "src/helpers/carousel/controller.js",
    "tests/helpers/carouselController.test.js"
  ],
  "outputs": [
    "src/helpers/carousel/controller.js",
    "src/helpers/carousel/keyboard.js",
    "src/helpers/carousel/swipe.js",
    "src/helpers/carousel/scrollSync.js",
    "src/helpers/carousel/resize.js",
    "tests/helpers/carouselController.test.js",
    "docs/components.md"
  ],
  "success": [
    "npx prettier . --check",
    "npx eslint .",
    "npm run check:jsdoc",
    "npx vitest run",
    "npx playwright test",
    "npm run check:contrast"
  ],
  "errorMode": "ask_on_public_api_change"
}
```

### Files Changed
- `src/helpers/carousel/controller.js`: delegate event wiring and centralize listener cleanup.
- `src/helpers/carousel/keyboard.js`: add keyboard navigation wiring helper.
- `src/helpers/carousel/swipe.js`: add touch/pointer swipe wiring helper.
- `src/helpers/carousel/scrollSync.js`: keep controller page state in sync with scroll.
- `src/helpers/carousel/resize.js`: debounce window resize to recompute metrics.
- `tests/helpers/carouselController.test.js`: test keyboard and swipe navigation.
- `docs/components.md`: document carousel controller API.

### Verification
- `npx prettier . --check`
- `npx eslint .`
- `npm run check:jsdoc` *(fails: missing or incomplete JSDoc blocks)*
- `npx vitest run`
- `npx playwright test` *(fails: 3 failing tests)*
- `npm run check:contrast`

### Risks
Refactored event wiring may have undiscovered edge cases; follow-up needed for failing Playwright specs and global JSDoc coverage.

------
https://chatgpt.com/codex/tasks/task_e_68bc926e545c8326a45187ff5fc1ee8f

### CODE REVIEW

File: src/helpers/carousel/scrollSync.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      if (ctrl._suppressScrollSync) return;
+      if (ctrl.metrics.pageWidth <= 0) {
+        ctrl.metrics = getPageMetrics(ctrl.container);
+      }
+      const { pageWidth: pw, pageCount: pc } = ctrl.metrics;
+      if (pw <= 0) return;
+      const page = Math.round(ctrl.container.scrollLeft / pw);
+      ctrl.currentPage = Math.max(0, Math.min(page, pc - 1));
+      ctrl.update();
+    });
```
[the remark this answers is not in the document]
Comment (CyanAutomation) [reply src/helpers/carousel/scrollSync.js:41]: @codex fix comments
